### PR TITLE
Add code samples to homepage

### PR DESCRIPTION
**Pull request by blaix**
Adds this to the bottom of the homepage:

Screenshot 2024-12-17 at 5 38 25 PM

### CODE REVIEW

File: src/Page/Index.elm [hunk 1]
```diff
@@ -98,5 +98,61 @@ view _ _ _ =
...
+                    ]
+                ]
+            , Html.p []
+                [ Html.text "In the "
+                , Html.a 
+                    [ Attribute.href "https://packages.gren-lang.org/package/gren-lang/browser/version/latest/overview" ]
+                    [ Html.text "browser:" ]
+                ]
+            , Html.pre []
+                [ Html.code []
```
Comment (robinheghan) [inline src/Page/Index.elm:107]: Does it need to be `<pre>` followed by `<code>`? 🤔

File: src/Page/Index.elm [hunk 2]
```diff
@@ -98,5 +98,61 @@ view _ _ _ =
             , Html.p []
                 [ Html.text "Gren produces small JavaScript files, and runs surprisingly fast. Applications written in Gren can be both smaller and faster than your average React-based application." ]
             ]
+        , Html.article
```
Comment (robinheghan) [inline src/Page/Index.elm]: Move this above `Compiler as an assistant` and `Efficiency`? We want to show people code as early as possible.

File: src/Page/Index.elm (hunk 1 above)
Comment (blaix) [reply src/Page/Index.elm:107]: Good question.

I think it is pretty typical to use both in this case. `pre` being more of a layout element and `code` more for semantics about what is in the preformatted block.

MDN says "`<pre>` elements commonly contain `<code>`, `<samp>`, and `<kbd>` elements, to represent computer code, computer output, and user input, respectively."

Looking at the package site, looks like code samples there are rendered with `<pre><code class="language-elm">`

Comment (blaix) [reply src/Page/Index.elm:107]: Well, as I just discovered pre + code is also required for the styling to work as-is. So I'll be doing that even if the contents aren't code 😅

**Comment by blaix** [comment]
Added error message to the compiler-as-assistant section:

Screenshot 2024-12-18 at 6 28 07 AM

**Review by robinheghan — APPROVED** [review]

**Pull request merged**